src/database/migrations/20220920054800-create-launches.js

```diff
@@ -0,0 +1,47 @@
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.createTable("launches", {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER,
+      },
+      user_id: {
+        allowNull: false,
+        type: Sequelize.INTEGER,
+      },
+      site_id: {
+        type: Sequelize.INTEGER,
+        allowNull: false,
+      },
+      primary_domain_source: {
+        allowNull: false,
+        type: Sequelize.STRING,
+      },
+      primary_domain_target: {
+        allowNull: false,
+        type: Sequelize.STRING,
+      },
+      domain_validation_source: {
+        allowNull: false,
+        type: Sequelize.STRING,
+      },
+      domain_validation_target: {
+        allowNull: false,
+        type: Sequelize.STRING,
+      },
```

Comment:
i don't think i have enough context here! could i get your help in explaining what `domain_validation_<source | target>` means?

i noticed your sequence diagram in the design doc has a domain validation lambda, but i couldn't match source/target to what the lambda does as the lambda should only require the agency's desired domain, no?

Reply:
sure! for every record (A, CNAME) that we store, we would have a source and a target. For clarity, in our system, we have 
www.blah.gov.sg CNAME gibberish.acm.aws

As such, our 
`domain_validation_source` = `www.blah.gov.sg` 
`domain_validation_target` = `gibberish.acm.aws`

For both the **_primary domains_** and the _**domain validations**_, we dont have to store the _type_ of record as it is guaranteed to be a CNAME record based on our architecture. 
The lambdas check to see if such a record has propagated

Reply:
yeap it does! why'd we need to store the `source` if we're pinging amplify for the status? is it just to make sure that the target -> source resolution is working?

Reply:
Arh - good question! This is because Amplify replies to us if the website is read to serve, but this does not guarantee that there is indeed a source -> target CNAME record that has actually propagated.

Reply:
ah ok so this is used later on in the services for verification of the source -> target CNAME right? if so, please feel free to mark this as resolved and thanks for answering!